x/programs/rust/examples/nft/src/lib.rs

```diff
@@ -0,0 +1,141 @@
+//! A basic ERC-721 compatible contract.
+//! The program serves as a non-fungible token with the ability to mint and burn.
+//! Only supports whole units with no decimal places.
+//!
+//! The NFT must support the common NFT metadata format.
+//! This includes the name, symbol, and URI of the NFT.
+use metadata::Nft;
+use wasmlanche_sdk::{program::Program, public, state_keys, types::Address};
+
+pub mod example;
+pub mod metadata;
+
+const NAME: &str = "My NFT";
+const SYMBOL: &str = "MNFT";
+const TOTAL_SUPPLY: u64 = 20;
+
+/// The program storage keys.
+#[state_keys]
+enum StateKey {
+    /// The total supply of the token. Key prefix 0x0.
+    TotalSupply,
+    /// The name of the token. Key prefix 0x1.
+    Name,
+    /// The symbol of the token. Key prefix 0x2.
+    Symbol,
+    /// Metadata of the token. Key prefix 0x3.
+    Metadata,
+    /// Balance of the NFT token by address. Key prefix 0x5(address).
+    Balance(Address),
+    /// Counter -- used to keep track of total NFTs minted.
+    Counter,
+}
+
+/// Initializes the NFT with all required metadata.
+/// This includes the name, symbol, image URI, owner, and total supply.
+/// Returns true if the initialization was successful.
+#[public]
+pub fn init(program: Program) -> bool {
+    // Set token name
+    program
+        .state()
+        .store(StateKey::Name.to_vec(), &NAME.as_bytes())
```

Comment:
I changed this because I wanted every state transition to be serial and explicit. We have talked about batching these but I think it needs more thought from a security perspective. I thought your way looked better but supporting it felt awkward.

Reply:
@hexfusion decided to break everything lol